webapp/niapp.yaml

```diff
@@ -0,0 +1,416 @@
+      operationId: get-content-index
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+      responses:
+        200:
+          description: The webapp's content. JSON for Dashboards, HTML for WebVIs
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+    put:
+      tags:
+        - content
+      summary: Upload content for an existing webapp
+      description: Upload content an existing webapp
+      operationId: update-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - in: body
+          required: true
+          name: content
+          description: The webapp content to upload
+          schema:
+            $ref: '#/definitions/WebAppContent'
+      responses:
+        200:
+          description: Success
+        400:
+          $ref: '#/responses/ValidationError'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+  '/api/webapps/{id}/content/{path}':
+    get:
+      tags:
+        - content
+      summary: Get webapp content by path
+      description: Get webapp content at path. For webapps with multiple files (e.g. WebVIs)
+      operationId: get-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - $ref: '#/parameters/WebAppContentPath'
+      responses:
+        200:
+          $ref: '#/definitions/WebAppContent'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+
+  '/webapps/{id}/duplicate/{sourceId}':
+    post:
+      tags:
+        - content
+      summary: Duplicates webapp content
+      description: Duplicates webapp content
+      operationId: duplicate-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - in: path
+          type: string
+          required: true
+          name: sourceId
+          description: The id of the source webapp to get copy content from
+      responses:
+        200:
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+
+parameters:
+  WebAppId:
+    in: path
+    name: id
+    description: The webapp identifier
+    type: string
+    required: true
+  WebAppContentPath:
+    in: path
+    name: path
+    description: Path to webapp content
+    type: string
+    required: true
+  CreateWebAppRequest:
+    in: body
+    name: CreateWebAppRequest
+    description: Create WebApp Request
+    schema:
+      type: object
+      title: Create WebApp Request
+      properties:
+        name:
+          type: string
+          description: The webapp's name
+        policyIds:
+          type: array
+          items:
+            type: string
+            example: "asdsad-17a6-45323-b64b-65325287372d"
+          description: List of policy Ids associated with the webapp, which give it access to the user's resources"
+        properties:
```

Comment:
Do we want to consider adding keywords as well, to match Tag and other datatypes?

Reply:
This is the documentation of the existing implementation.